src/Microsoft.OData.Client/Materialization/MaterializerDeltaResourceSet.cs

```diff
@@ -0,0 +1,97 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="MaterializerDeltaResourceSet.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.OData.Client.Materialization
+{
+    /// <summary>
+    /// Materializer state for a given <see cref="ODataDeltaResourceSet"/>.
+    /// </summary>
+    internal class MaterializerDeltaResourceSet : IMaterializerState
+    {
+        /// <summary>The OData delta resource set.</summary>
+        private readonly ODataDeltaResourceSet deltaResourceSet;
+
+        /// <summary>The entries.</summary>
+        private readonly List<IMaterializerState> entries;
+
+        /// <summary>
+        /// Prevents a default instance of the <see cref="MaterializerDeltaResourceSet"/> struct from being created.
+        /// </summary>
+        /// <param name="deltaResourceSet">The OData delta resource set.</param>
+        /// <param name="entries">The entries.</param>
+        private MaterializerDeltaResourceSet(ODataDeltaResourceSet deltaResourceSet, List<IMaterializerState> entries)
+        {
+            Debug.Assert(deltaResourceSet != null, "deltaResourceSet != null");
+            Debug.Assert(entries != null, "entries != null");
+
+            this.deltaResourceSet = deltaResourceSet;
+            this.entries = entries;
+        }
+
+        /// <summary>
+        /// Gets the delta resource set.
+        /// </summary>
+        public ODataDeltaResourceSet DeltaResourceSet
+        {
+            get { return this.deltaResourceSet; }
+        }
+
+        /// <summary>
+        /// Gets the entries.
+        /// </summary>
+        public List<IMaterializerState> Entries
+        {
+            get { return this.entries; }
+        }
+
+        /// <summary>
+        /// Adds a <see cref="IMaterializerState"/> entry to the entries collection.
+        /// </summary>
+        /// <param name="entry">The <see cref="IMaterializerState"/> entry to be added.</param>
+        public void AddEntry(IMaterializerState entry)
+        {
+            this.entries.Add(entry);
+        }
+
+        /// <summary>
+        /// Creates the delta resource set.

```

Comment:
Seems like the delta resource set is the input to this method, so what is actually being created?